paddle/fluid/framework/op_registry.h

```diff
@@ -149,16 +153,29 @@ class OpKernelRegistrar : public Registrar {
 /**
  * Macro to register OperatorKernel.
  */
-#define REGISTER_OP_KERNEL(op_type, LIBRARY_TYPE, place_class, ...)        \
-  STATIC_ASSERT_GLOBAL_NAMESPACE(                                          \
-      __reg_op_kernel_##op_type##_##LIBRARY_TYPE##__,                      \
-      "REGISTER_OP_KERNEL must be called in global namespace");            \
-  static ::paddle::framework::OpKernelRegistrar<place_class, __VA_ARGS__>  \
-      __op_kernel_registrar_##op_type##_##LIBRARY_TYPE##__(#op_type,       \
-                                                           #LIBRARY_TYPE); \
-  int TouchOpKernelRegistrar_##op_type##_##LIBRARY_TYPE() {                \
-    __op_kernel_registrar_##op_type##_##LIBRARY_TYPE##__.Touch();          \
-    return 0;                                                              \
+#define REGISTER_OP_KERNEL(op_type, LIBRARY_TYPE, place_class, ...)       \
+  STATIC_ASSERT_GLOBAL_NAMESPACE(                                         \
+      __reg_op_kernel_##op_type##_##LIBRARY_TYPE##__,                     \
+      "REGISTER_OP_KERNEL must be called in global namespace");           \
+  static ::paddle::framework::OpKernelRegistrar<place_class, __VA_ARGS__> \
+      __op_kernel_registrar_##op_type##_##LIBRARY_TYPE##__(               \
+          #op_type, #LIBRARY_TYPE, "ANYLAYOUT");                          \
+  int TouchOpKernelRegistrar_##op_type##_##LIBRARY_TYPE() {               \
+    __op_kernel_registrar_##op_type##_##LIBRARY_TYPE##__.Touch();         \
+    return 0;                                                             \
+  }
+
+#define REGISTER_OP_KERNEL_WITH_LAYOUT(op_type, LIBRARY_TYPE, LAYOUT,        \
```

Comment:
Why do you `REGISTER_OP_KERNEL_WITH_LAYOUT`? I could not find the usage of it.

Reply:
According to your previous proposition about shares the pull-request into a few small parts of the code, this one contains only changes required to support MKLDNN operators, but it doesn't contains example how this #define is uses. Could you have a look at this code ([#usage](https://github.com/pzelazko-intel/Paddle/blob/9b84444f11a146f7536b121f4b68b8a4d66d8af5/paddle/fluid/operators/conv_mkldnn_op.cc#L374))